ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/BlockTree.hs

```diff
@@ -75,3 +88,84 @@ slotLength fragment =
   fromIntegral $ unSlotNo $
     withOrigin 0 id (AF.anchorToSlotNo $ AF.headAnchor fragment)
     - withOrigin 0 id (AF.anchorToSlotNo $ AF.anchor fragment)
+
+-- | Return all the full fragments from the root of the tree.
+allFragments :: BlockTree blk -> [AF.AnchoredFragment blk]
+allFragments BlockTree{..} = trunk : map full branches
+
+firstJust :: [Maybe a] -> Maybe a
+firstJust [] = Nothing
+firstJust (Nothing : rest) = firstJust rest
+firstJust (Just x : _) = Just x
```

Comment:
[Data.Foldable.asum](https://hackage.haskell.org/package/base-4.18.1.0/docs/Data-Foldable.html#v:asum)?

Reply:
Looks like it. I was sure it would exist somewhere but I couldn't find it. Thanks! Done in 49ec7d700.